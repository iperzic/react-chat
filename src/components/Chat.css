.Chat {
    min-height: 100%;
    background: url('./assets/Body BG.png') no-repeat fixed;
    background-size: cover;
}

.Chat__last-message-handle {
    visibility: hidden;
}
