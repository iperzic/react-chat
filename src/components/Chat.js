import React from 'react';
import PropTypes from 'prop-types';

import './Chat.css';

import MessageBubble from './MessageBubble';
import NewMessage from './NewMessage';

class Chat extends React.Component {
  componentDidMount() {
    this.props.onChatLoad();
  }

  render() {
    return (
      <div className="Chat">
        <div className="Chat__messages">
          {this.props.messages.map(m =>
            (<MessageBubble
              author={m.author}
              message={m.message}
              timestamp={m.timestamp}
              isOwn={m.isOwn}
            />))
          }
        </div>
        <div className="Chat__new-message-bar">
          <NewMessage onMessageSend={this.props.onNewMessage} />
        </div>
      </div>
    );
  }
}

Chat.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape({
    author: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    timestamp: PropTypes.string.isRequired,
    isOwn: PropTypes.bool.isRequired,
  })).isRequired,
  onNewMessage: PropTypes.func.isRequired,
  onChatLoad: PropTypes.func.isRequired,
};

export default Chat;
