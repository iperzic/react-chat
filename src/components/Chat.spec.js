import React from 'react';
import { shallow } from 'enzyme';

import Chat from './Chat';

describe('Chat', () => {
  let component;
  let messageSendSpy;
  let chatLoadSpy;
  let messages;

  beforeEach(() => {
    messageSendSpy = jest.fn();
    chatLoadSpy = jest.fn();
    messages = [{
      id: 'abc', author: 'Bugs Bunny', message: 'Whats up doc?', timestamp: 'some timestamp', isOwn: true,
    }, {
      id: 'abcd', author: 'Foo', message: 'Bar', timestamp: 'Baz', isOwn: false,
    }];
    component = shallow(<Chat
      messages={messages}
      onNewMessage={messageSendSpy}
      onChatLoad={chatLoadSpy}
    />);
  });

  it('should display all provided messages with a new message', () => {
    expect(component.getElement()).toMatchSnapshot();
  });

  it('should call the onChatLoad prop when a component is mounted', () => {
    expect(chatLoadSpy).toHaveBeenCalled();
  });
});
