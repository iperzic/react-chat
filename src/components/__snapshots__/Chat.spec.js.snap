// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat should display all provided messages with a new message 1`] = `
<div
  className="Chat"
>
  <div
    className="Chat__messages"
  >
    <MessageBubble
      author="Bugs Bunny"
      isOwn={true}
      message="Whats up doc?"
      timestamp="some timestamp"
    />
    <MessageBubble
      author="Foo"
      isOwn={false}
      message="Bar"
      timestamp="Baz"
    />
  </div>
  <div
    className="Chat__new-message-bar"
  >
    <NewMessage
      onMessageSend={[Function]}
    />
  </div>
</div>
`;
